filebeat/prospector/prospector.go

```diff
@@ -128,17 +120,19 @@ func (p *Prospector) Start() {
 	p.wg.Add(1)
 	logp.Info("Starting prospector of type: %v; id: %v ", p.config.InputType, p.ID())
 
+	onceWg := sync.WaitGroup{}
 	if p.Once {
-		// Makes sure prospectors can complete first scan before stopped
-		defer p.runWg.Wait()
+		// Make sure start is only completed when Run did a complete first scan
+		defer onceWg.Wait()
 	}
 
+	onceWg.Add(1)
 	// Add waitgroup to make sure prospectors finished
-	p.runWg.Add(1)
 	go func() {
 		defer func() {
-			p.runWg.Done()
+			onceWg.Done()
```

Comment:
`Start` can return as soon as `Run()` is completed. As `Run` is only run once, it will be already stopping when Run is finished. So `onceWg.Done()` is ok to be called before `p.stop()`.